include/deal.II/lac/la_parallel_vector.templates.h

```diff
@@ -654,6 +995,29 @@ namespace LinearAlgebra
         import_data =
           std_cxx14::make_unique<Number[]>(partitioner->n_import_indices());
 
+#  if defined(DEAL_II_WITH_CUDA) && defined(__CUDACC__)
+      // TODO: for now move the data to the host and then move it back to the
+      // the device. We use values to store the elements because the function
+      // uses a view of the array and thus we need the data on the host to
+      // outlive the scope of the function.
+      if (std::is_same<MemorySpace, CUDA>::value)
+        {
+          Number *new_val;
+          Utilities::System::posix_memalign((void **)&new_val,
+                                            64,
+                                            sizeof(Number) * allocated_size);
+
+          values.reset(new_val);
```

Comment:
Ah, here you currently need both arrays.

Reply:
We could still store such a pointer in the `CUDA` memory space (until we use `CUDA`-aware `MPI`).